### PR TITLE
Add no-cache option

### LINKED ISSUE #165: need option "--no-cache" to force docker for clean build of an image.
There is no docker build option "--no-cache" to force docker for clean build of an image.

Could we have this feature?

```
...
Step 3 : RUN apt-get update
 ---> Using cache
 ---> bcbca2fcf204
Step 4 : RUN apt-get install -y openjdk-7-jdk
 ---> Using cache
 ---> 103f1a261d44
Step 5 : RUN apt-get install -y openssh-server
 ---> Using cache
 ---> dde41f8d0904
Step 6 : RUN apt-get install -y git-core
 ---> Using cache
 ---> 9be002f08b6a
Step 7 : RUN apt-get install -y build-essential
 ---> Using cache
 ---> a752fd73a698
```

### PR DESCRIPTION
**Pull request by shovelwagon**
Allows for clean build using --no-cache. Resolves #165

### CODE REVIEW

**Comment by shovelwagon** [comment]
@bradrydzewski or @tboerger, wondering if one of you could review?

File: cmd/drone-docker/main.go [hunk 1]
```diff
@@ -202,6 +202,11 @@ func main() {
 			Usage:  "repository default branch",
 			EnvVar: "DRONE_REPO_BRANCH",
 		},
+		cli.BoolFlag{
+			Name:   "no-cache",
+			Usage:  "donot use cached intermediate containers",
+			EnvVar: "NO_CACHE",
```
[the remark this answers is not in the document]
Comment (shovelwagon) [reply cmd/drone-docker/main.go]: Ah, yep. Thanks. This is fixed.

**Comment by shovelwagon** [comment]
@bradrydzewski, proposed change has been completed.

**Comment by ozbillwang** [comment]
Thanks, @shovelwagon 

Could you also update the related document at http://plugins.drone.io/drone-plugins/drone-docker/ that now we have a new option to be used.

**Pull request merged**